rialto_airflow/harvest/distill.py

```diff
@@ -203,12 +204,18 @@ def _first(pub, rules: Rules) -> Optional[str | int]:
             results = jpath.find(data)
             if len(results) > 0:
                 value = results[0].value
-                if rule.is_int:
+                if rule.is_valid_year:
                     try:
-                        return int(value)
+                        year_val = int(value)
+                        if year_val <= datetime.datetime.now().year:
```

Comment:
Maybe a question for @jacobthill: would we want to loosen this rule just a little to allow one year beyond the current? 2026 publications will likely be showing up in our sources in late 2025. We would not get them for our reports until we run a harvest in 2026. This depends on what you want to have available in the dashboards.

Reply:
There is currently logic for determining the pub_year: 

https://github.com/sul-dlss/rialto-airflow/blob/c847a589a70e2f995c56792706e9428969cee4c4/rialto_airflow/harvest/distill.py#L73-L87

Basically to look in sul_pub, dimensions, openalex, and then web-of-science. This PR would mean that it would check to make sure the pub_year is less than or equal to the current year, and would be applied in that order as well. So if sul_pub has 2026 and dimensions has 2025, we will use 2025.

One advantage to applying the limit at query time is that we can change it easily without needing to reharvest. We could also get a picture of what the publication years look like.